cypress-tests/cypress/support/commands.js

```diff
@@ -141,11 +141,20 @@ Cypress.Commands.add(
     // it is best to use then() to handle the response within the same block of code
     cy.readFile(globalState.get("connectorAuthFilePath")).then(
       (jsonContent) => {
-        const authDetails = getValueByKey(
+        let authDetails = getValueByKey(
           JSON.stringify(jsonContent),
-          `${connectorName}_payout`,
+          `${connectorName}_payout`
         );
 
+        let sourceBalanceAccount = null;
+        if (authDetails && typeof authDetails === "object") {
+          // Specific to `AdyenPlatform` payout connector
+          // If the connector has source_balance_account in creds file, set it in the body and remove it from authDetails
+          sourceBalanceAccount = authDetails.source_balance_account;
+          delete authDetails.source_balance_account;
+        }
+        authDetails = authDetails;
+
```

Comment:
Yes, this was hardcoded and is specific to `adyenplatform` connector (mentioned in comments as well) as `sourceBalanceAccount` is being passed as a value to `metadata`. It is a secret value which cannot be made public (can be treated as `key1`)

For context, above is a temporary change and I'll be modifying `get_value_by_key` to accommodate below mentioned change:

- `get_value_by_key()` has been duplicated in multiple places, this can be unified
  - Can be 2 copies, one with merger (logic that merges `common.js` and `connector.js` files into one) and the other one being the legacy one. Will look into it
- format of creds will have to be modified first:
	```js
	// from
	{
		"connector": {
			"auth_type": "KeyType",
			"api_key": "api_key",
			"key1": "key1",
			// other kv pairs
		},
		// another one
	}
	// to something like, will be decided later when picked up
	{
		"connector": {
			"auth_type": "KeyType",
			"api_key": "api_key",
			"key1": "key1",
			// other kv pairs
			"metadata": {
				"key": "value",
				// other kv pairs
			}
		},
	}
	```
- Any `key` that lies outside of `auth_info` (`auth_type`, `api_key`, `key1`, `key2`, and `api_secret`) should be treated as `metadata`